components/vc/front/src/torque.c

```diff
@@ -84,19 +129,78 @@ CAN_torqueManagerState_E torque_getStateCAN(void)
     return ret;
 }
 
+/**
+ * @brief Get current race mode
+ * @return State of the torque manager race mode
+ */
+torque_raceMode_E torque_getRaceMode(void)
+{
+ return torque_data.race_mode;
+}
+
+/**
+ * @brief Translate race mode to CAN
+ * @return CAN state of the torque manager race mode
+ */
+CAN_raceMode_E torque_getRaceModeCAN(void)
+{
+    CAN_raceMode_E ret = CAN_RACEMODE_PIT;
+
+    switch (torque_data.race_mode)
+    {
+        case RACEMODE_ENABLED:
+            ret = CAN_RACEMODE_RACE;
+            break;
+        default:
+            break;
+    }
+
+    return ret;
+}
+
 static void torque_init(void)
 {
     memset(&torque_data, 0x00U, sizeof(torque_data));
 
     torque_data.state = TORQUE_INACTIVE;
-    torque_data.torque_request_max = DEFAULT_TORQUE_PITS;
+    torque_data.torque_request_max = DEFAULT_BOOT_TORQUE;
 }
 
 static void torque_periodic_100Hz(void)
 {
-    CAN_raceMode_E race_mode = CAN_RACEMODE_PIT;
+    CAN_digitalStatus_E gear_change_request = CAN_DIGITALSTATUS_SNA;
+    CAN_digitalStatus_E race_mode_change_requested = CAN_DIGITALSTATUS_SNA;
+    float32_t torque_request_max = torque_data.torque_request_max;
     float32_t torque = 0.0f;
-    const bool race_mode_not_sna = CANRX_get_signal(VEH, STW_raceMode, &race_mode) != CANRX_MESSAGE_SNA;
+
+    const bool gear_change_was_requested = torque_data.gear_change_active;
+    torque_data.gear_change_active = (CANRX_get_signal(VEH, SWS_requestReverse, &gear_change_request) != CANRX_MESSAGE_SNA) &&
+                                     (gear_change_request == CAN_DIGITALSTATUS_ON);
+    const bool gear_change_rising = !gear_change_was_requested && torque_data.gear_change_active;
+
+    const bool race_mode_change_was_requested = torque_data.race_mode_change_active;
+    torque_data.race_mode_change_active = (CANRX_get_signal(VEH, SWS_requestRaceMode, &race_mode_change_requested) != CANRX_MESSAGE_SNA) &&
+                                          (race_mode_change_requested == CAN_DIGITALSTATUS_ON);
+    const bool race_mode_change_rising = !race_mode_change_was_requested && torque_data.race_mode_change_active;
+
+    CAN_digitalStatus_E torque_change_request = CAN_DIGITALSTATUS_SNA;
+    const bool torque_inc_active = (CANRX_get_signal(VEH, SWS_requestTorqueInc, &torque_change_request) != CANRX_MESSAGE_SNA) &&
```

Comment:
how does `SWS_requestTorqueInc`/`Dec` work? Are you increasing/decreasing by a certain set point each time? If so, I might suggest just having an enum for different "torque modes", 0 through 15 or something. Then map those to specific torque values on the receive side.